### PR TITLE
Feat: add user management apis

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

### Description of your changes

 Add user management apis

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

/cc @barnettZQG

### CODE REVIEW

File: pkg/apiserver/model/user.go [hunk 1]
```diff
@@ -54,6 +59,37 @@ func (u *User) Index() map[string]string {
 	return index
 }
 
+// ProjectUser is the model of user in project
+type ProjectUser struct {
+	BaseModel
+	Username     string `json:"username"`
+	ProjectName  string `json:"projectName"`
+	ProjectAlias string `json:"projectAlias,omitempty"`
+	UserRole     string `json:"userRole"`
```
Comment (barnettZQG) [inline pkg/apiserver/model/user.go:68]:
```suggestion
	UserRoles    []string `json:"userRoles"`
```

File: pkg/apiserver/model/user.go [hunk 2]
```diff
@@ -54,6 +59,37 @@ func (u *User) Index() map[string]string {
 	return index
 }
 
+// ProjectUser is the model of user in project
+type ProjectUser struct {
+	BaseModel
+	Username     string `json:"username"`
+	ProjectName  string `json:"projectName"`
+	ProjectAlias string `json:"projectAlias,omitempty"`
```
Comment (barnettZQG) [inline pkg/apiserver/model/user.go:67]: Repeated recording of the project alias can lead to inconsistencies?

File: pkg/apiserver/rest/apis/v1/types.go [hunk 3]
```diff
@@ -1088,17 +1090,62 @@ type LoginResponse struct {
...
+	UserRole string `json:"userRole"`
+}
+
+// CreateUserRequest create user request
+type CreateUserRequest struct {
+	Name  string `json:"name" validate:"checkname"`
+	Alias string `json:"alias,omitempty" validate:"checkalias" optional:"true"`
+	Email string `json:"email" validate:"checkemail"`
+	// TODO: add password validation
+	Password string `json:"password"`
```
Comment (barnettZQG) [inline pkg/apiserver/rest/apis/v1/types.go:1120]: add the password validate

File: pkg/apiserver/rest/webservice/user.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Writes(apis.ListUserResponse{}))
+
+	ws.Route(ws.POST("/").To(c.createUser).
+		Doc("create a user").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Reads(apis.CreateUserRequest{}).
+		Returns(200, "OK", apis.DetailUserResponse{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Writes(apis.DetailUserResponse{}))
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/user.go:71]: return the DetailUserResponse but the projects do not set values, inaccurate for generated API documentation.

File: pkg/apiserver/rest/webservice/user.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+// NewUserWebService is the webservice of user
+func NewUserWebService(userUsecase usecase.UserUsecase) WebService {
+	return &userWebService{
+		userUsecase: userUsecase,
+	}
+}
+
+func (c *userWebService) GetWebService() *restful.WebService {
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/user.go:43]: Missing an API to get info about a logged-in user?

File: pkg/apiserver/rest/webservice/webservice.go [hunk 6]
```diff
@@ -100,6 +101,7 @@ func Init(ds datastore.DataStore, addonCacheTime time.Duration) {
 
 	// Authentication
 	RegisterWebService(NewAuthenticationWebService(authenticationUsecase))
+	RegisterWebService(NewUserWebService(userUsecase))
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/webservice.go:104]: Can you add some e2e test cases?

File: pkg/apiserver/rest/apis/v1/types.go (hunk 3 above)
Comment (FogDong) [reply pkg/apiserver/rest/apis/v1/types.go:1120]: Not sure of the validation, is there any rule for it?

File: pkg/apiserver/rest/webservice/user.go (hunk 5 above)
Comment (FogDong) [reply pkg/apiserver/rest/webservice/user.go:43]: We can discuss that in the next PR.

File: pkg/apiserver/rest/apis/v1/types.go [hunk 7]
```diff
@@ -1088,17 +1090,61 @@ type LoginResponse struct {
...
+	Name     string `json:"name" validate:"checkname"`
+	Alias    string `json:"alias,omitempty" validate:"checkalias" optional:"true"`
+	Email    string `json:"email" validate:"checkemail"`
+	Password string `json:"password" validate:"checkpassword"`
+}
+
+// UpdateUserRequest update user request
+type UpdateUserRequest struct {
+	Alias    string `json:"alias,omitempty" optional:"true"`
+	Password string `json:"password,omitempty" validate:"checkpassword" optional:"true"`
```
Comment (barnettZQG) [inline pkg/apiserver/rest/apis/v1/types.go:1125]: The password and email can be empty, but the checkpassword and checkemail validate function can not pass.

**Review by barnettZQG — APPROVED** [review]
LGTM after fixing the test error.

**Pull request merged**